x/upgrade/keeper/keeper.go

```diff
@@ -185,9 +185,9 @@ func (k Keeper) GetModuleVersions(ctx context.Context) ([]*types.ModuleVersion,
 	prefix := []byte{types.VersionMapByte}
 	it, err := store.Iterator(prefix, storetypes.PrefixEndBytes(prefix))
 	if err != nil {
-		_ = it.Close()
 		return nil, err
 	}
+	defer it.Close()
```

Comment:
if it errors above is this still run?

Reply:
No, it returns the error immediately, skipping the defer statement.

Reply:
Static check complains about it with `should check returned error before deferring it.Close() (SA5001)`. I suppose it makes sense because if it's returning an error then the iterator never got opened

Reply:
that makes sense.
Then in this case maybe we need to handle err returned from `it.Close()` since there is a possibility of errors coming up while closing like resource availability, connection errors etc.
Maybe something like this:
```defer func() {
    if err := it.Close(); err != nil {
        // handle the err/log
    }
}()
```

Reply:
We've been ignoring these errors all over the SDK, would like to fix them in a follow up PR instead of adding that to this one